x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/ExpressionBuilder.java

```diff
@@ -672,6 +677,33 @@ public List<Alias> visitFields(EsqlBaseParser.FieldsContext ctx) {
         return ctx != null ? visitList(this, ctx.field(), Alias.class) : new ArrayList<>();
     }
 
+    @Override
+    public NamedExpression visitAggField(EsqlBaseParser.AggFieldContext ctx) {
+        Source source = source(ctx);
+        Alias field = visitField(ctx.field(), source);
+        var filterExpression = ctx.booleanExpression();
+        Expression condition = filterExpression != null ? expression(filterExpression) : null;
+
+        if (condition != null) {
```

Comment:
```suggestion

        if (filterExpression != null) {
            Expression condition = expression(filterExpression);
```

expression() will throw, rather than return `null`.